proto/viam/common/v1/common.proto

```diff
@@ -15,6 +15,7 @@ message ResourceName {
   string type = 2;
   string subtype = 3;
   string name = 4;
+  optional string machine_part_id = 5;
```

Comment:
Do we not also want the machine ID? Or just the part ID?

Reply:
just part id for now, that's what was scoped and most likely to be useful - user can get cloud metadata if they need more of that stuff